src/storage/heap_file.c

```diff
@@ -25091,19 +25091,70 @@ heap_get_visible_version_with_repl_desync (THREAD_ENTRY * thread_p, HEAP_GET_CON
 	}
       assert (is_passive_transaction_server ());
       // Handle the page desynchronization error
-      if (context->home_page_watcher.pgptr)
-	{
-	  /* Unfix home page. */
-	  pgbuf_ordered_unfix (thread_p, &context->home_page_watcher);
-	}
 
-      if (context->fwd_page_watcher.pgptr != NULL)
+      // clean - saving state if required
+      //
+      heap_clean_get_context (thread_p, context);
+
+      // also unfix what is in cache watcher; saving the state for a future re-fix after the wait
+      //
+      VPID saved_vpid VPID_INITIALIZER;
+      PGBUF_LATCH_MODE saved_latch_mode = PGBUF_NO_LATCH;
+      bool restore_scan_cache_page_watcher = false;
+      if (is_heap_scan)
```

Comment:
Just a question, 
I don't understand why this condition (`is_heap_scan`) is required when unfix the page buffer. It would be very helpful if you could explain why.

Reply:
You might be right.
I also checked and there is no apparent link between `is_heap_scan` and the fact that the home page watcher is saved in `scan_cache`.
The save in `scan_cache` happens in `heap_get_visible_version_internal` -> `heap_prepare_get_context` -> `heap_clean_get_context`.
Removed the check.